mlflow/types/type_hints.py

```diff
@@ -261,7 +302,7 @@ def _validate_example_against_type_hint(example: Any, type_hint: type[Any]) -> A
             return type_hint(**example_dict) if isinstance(example, dict) else example
     elif type_hint == Any:
         return example
-    elif type_hint in TYPE_HINTS_TO_DATATYPE_MAPPING:
+    elif type_hint in TYPE_HINTS_TO_DATATYPE_MAPPING or type_hint in no_op_type_hints():
```

Comment:
Should we do validation for no-op type hints? For example, users might put `pd.DataFrame` as type hint but give a dictionary input example as they know MLflow would convert it to pandas DataFrame internally. If we claim those to "no-op", shall we skip input example validation here? Since we can think of many edge cases here (e.g. nested dataframe), hard-blocking here does not seem to be safe bet for us.

Reply:
Yea seems this function won't be invoked on such type hints anyways, let me remove it